Tasks/DeleteFilesV1/Tests/L0.ts

```diff
@@ -0,0 +1,118 @@
+        this.timeout(5000);
+
+        const root = path.join(testRoot, 'nested');
+        fs.mkdirSync(root);
+
+        fs.mkdirSync(path.join(root, 'A'));
+        fs.writeFileSync(path.join(root, 'A', 'test.txt'), 'test');
+        fs.mkdirSync(path.join(root, 'A', 'A'));
+        fs.writeFileSync(path.join(root, 'A', 'A', 'test2.txt'), 'test2');
+        fs.writeFileSync(path.join(root, 'A', 'A', 'test3.txt'), 'test3');
+        fs.mkdirSync(path.join(root, 'B'));
+        fs.writeFileSync(path.join(root, 'B', 'test4.txt'), 'test4');
+        fs.mkdirSync(path.join(root, 'C'));
+        fs.writeFileSync(path.join(root, 'C', 'dontDelete.txt'), 'dont delete');
+
+        let tp: string = path.join(__dirname, 'L0Nested.js');
+        let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        runValidations(() => {
+            assert(!fs.existsSync(path.join(root, 'A')));
+            assert(!fs.existsSync(path.join(root, 'B')));
+            assert(fs.existsSync(path.join(root, 'C')));
+            assert(fs.existsSync(path.join(root, 'C', 'dontDelete.txt')));
+        }, tr, done);
+    });
+
+    it('Deletes a single file', (done: MochaDone) => {
+        this.timeout(5000);
+
+        const root = path.join(testRoot, 'singleFile');
+        fs.mkdirSync(root);
+
+        fs.mkdirSync(path.join(root, 'A'));
+        fs.writeFileSync(path.join(root, 'A', 'test.txt'), 'test');
+        fs.mkdirSync(path.join(root, 'A', 'A'));
+        fs.writeFileSync(path.join(root, 'A', 'A', 'test.txt'), 'test2');
+        fs.writeFileSync(path.join(root, 'A', 'A', 'test2.txt'), 'test3');
+
+        let tp: string = path.join(__dirname, 'L0SingleFile.js');
+        let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        runValidations(() => {
+            assert(fs.existsSync(path.join(root, 'A')));
+            assert(!fs.existsSync(path.join(root, 'A', 'test.txt')));
+            assert(fs.existsSync(path.join(root, 'A', 'A')));
+            assert(fs.existsSync(path.join(root, 'A', 'A', 'test.txt')));
+            assert(fs.existsSync(path.join(root, 'A', 'A', 'test2.txt')));
+        }, tr, done);
+    });
+
+    it('Removes the source folder if its empty', (done: MochaDone) => {
+        this.timeout(5000);
+
+        const root = path.join(testRoot, 'rmSource');
+        fs.mkdirSync(root);
+
+        fs.mkdirSync(path.join(root, 'A'));
+        fs.writeFileSync(path.join(root, 'A', 'test.txt'), 'test');
+        fs.mkdirSync(path.join(root, 'A', 'A'));
+        fs.writeFileSync(path.join(root, 'A', 'A', 'test2.txt'), 'test2');
+        fs.writeFileSync(path.join(root, 'A', 'A', 'test3.txt'), 'test3');
+
+        let tp: string = path.join(__dirname, 'L0RmSource.js');
+        let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        runValidations(() => {
+            assert(!fs.existsSync(root));
+        }, tr, done);
+    });
+});
```

Comment:
These are all happy path tests which is good. Are there any bad path tests we want to add in here? Like directories with locked down permissions? Or, what if we try to deleted "/" or ".." ?

Reply:
Added